deploy/helm/sumologic/conf/metrics/collector/otelcol/config.yaml

```
@@ -57,8 +57,8 @@ processors:
     metric_statements:
       - context: metric
         statements:
-          - extract_sum_metric(true) where type == METRIC_DATA_TYPE_HISTOGRAM or type == METRIC_DATA_TYPE_EXPONENTIAL_HISTOGRAM or type == METRIC_DATA_TYPE_SUMMARY
-          - extract_count_metric(true) where type == METRIC_DATA_TYPE_HISTOGRAM or type == METRIC_DATA_TYPE_EXPONENTIAL_HISTOGRAM or type == METRIC_DATA_TYPE_SUMMARY
+          - extract_sum_metric(true) where not IsMatch(name, .Values.sumologic.metrics.allowHistogramRegex) and type == METRIC_DATA_TYPE_HISTOGRAM or type == METRIC_DATA_TYPE_EXPONENTIAL_HISTOGRAM or type == METRIC_DATA_TYPE_SUMMARY
```

@sumo-drosiek seems like an allowlist (using regex) here is invalid, since ottl doesn't support template variables with regex pattern matching. The second argument to `IsMatch` needs to be a string.

---

This is helm template, so we should use the curly braces to interpolate the variable, like here:

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/8e3a9e765021467daa639e51ef65ea1efb058163/deploy/helm/sumologic/conf/metrics/collector/otelcol/config.yaml#L9

We can use goldenfiles to ensure that the template is rendered correctly

---

I would also use parenthesis to group remaining conditions to ensure that everything is resolved correctly

---

I understand that, however ottl doesn't seem to support using helm template variables or variables in general to specify regex patterns.

---

I don't understand. After template will be generated, there won't be any variables

```suggestion
          - extract_sum_metric(true) where not IsMatch(name, {{ .Values.sumologic.metrics.allowHistogramRegex | quote }}) and type == METRIC_DATA_TYPE_HISTOGRAM or type == METRIC_DATA_TYPE_EXPONENTIAL_HISTOGRAM or type == METRIC_DATA_TYPE_SUMMARY
```

---

ok let me test out the change.

---

I see what you mean, the value of the template variable is injected before evaluating the ottl expression. thanks!

---

for some reason the `_bucket` metrics are passing through with this change.

---

I think everything matches empty string. We should make it more sophisticated, so add `not IsMatch(name, {{ .Values.sumologic.metrics.allowHistogramRegex | quote }}) and ` only if `.Values.sumologic.metrics.allowHistogramRegex` is not empty